elf2flt.c

```diff
@@ -776,6 +776,12 @@ output_relocs (
 				case R_XTENSA_DIFF8:
 				case R_XTENSA_DIFF16:
 				case R_XTENSA_DIFF32:
+				case 57: /*R_XTENSA_PDIFF8*/
```

Comment:
won't these defines be provided by binutils ?

Reply:
They will. My reasoning was that they are a part of ABI now and won't change. On the contrary definitions are present in new binutils and are missing in older binutils, and they are not macros, they are enum, so there's no easy way to check whether they are defined or not short of adding a test to configure.
If that's preferred I can do that.

Reply:
we've been avoiding hardcoding reloc numbers since that gets pretty messy & easy to mess up.  the lack of defines for compile-time testing is unfortunate otherwise i'd say just bracket the statements with that.

our model has been that the version of binutils that elf2flt is compiled against is also the version it is run against.  so if the relocs don't exist at compile time, then they shouldn't show up at runtime when processing programs.

but hardcoding the symbolic names effectively requires a newer version of binutils if people want to build against xtensa.  i'm guessing you don't want to require that any time soon :).

another downside is that binutils doesn't expose a version for us to test against at compile time either, so we couldn't do (equiv of) "#if BFD_VER < 2.35".

i'd be fine with either:
* add a configure test for the relocs and defining something like `HAVE_BFD_XTENSA_DIFF_RELOCS` to key off of in elf2flt.c
* keep the list you've written here, but add a comment like "switch to the reloc symbols in Jan 2021" which would effectively require a newer binutils at that time